hcloud/hcloud.py

```diff
@@ -54,6 +61,13 @@ def _get_headers(self):
         }
         return headers
 
+    def with_endpoint(self, api_endpoint):
```

Comment:
Why do we need a `with_endpoint()` method? Why not use the `api_endpoint` property directly? It’s public anyway.

Reply:
We wanted to have similar configuration methods on the client like hcloud-go, so we have some of the `with_*`- Methods.

I guess the property `api_endpoint` shouldn't be public, we should have a setter (`with_endpoint`) and for the testing also a getter.
What do you think?

Reply:
In Go, we have the `With` prefix because we use the [functional options pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) for configuring the client.

I don’t see why we need a method here — a property is fine.

Reply:
After talking with @thcyron about it, i think it would be a better solution to use the `with_*` Methods for a fluent interface.
Sample:
```python
client.with_endpoint("bla").with_application("ansible","1.0.0")
 # instead of
client.api_endpoint = "bla"
client.application_name = "ansible"
client.application_version = "1.0.0"
```

And i guess we should make the properties `api_endpoint`, `application_*` protected.

Reply:
```suggestion
    client = HcloudClient(application_name="ansible")
```

Do we allow to change endpoint?

Reply:
Yes we do, it should be possible to change the endpoint to our internal testing systems (when developing new features as sample).